airflow/utils/cli.py

```diff
@@ -341,19 +346,30 @@ def _wrapper(*args, **kwargs):
     return cast(T, _wrapper)
 
 
-def get_config_with_source(include_default=False):
+def get_config_with_source(include_default=False, include_colors=True):
     """
     Return configuration along with source for each option.
     """
-    parser = AirflowConfigParser(strict=False, interpolation=None)
     config_dict = conf.as_dict(display_source=True)
 
     with io.StringIO() as buf, redirect_stdout(buf):
         for section, options in config_dict.items():
-            print(f"[{section}]")
-            for key, (value, source) in options.items():
-                if not include_default and source == "default":
-                    continue
-                print(f"{key} = {value} [{source}]")
-            print()
-        return buf.getvalue()
+            if not include_default:
+                options = {
+                    key: (value, source) for key, (value, source) in options.items() if source != "default"
+                }
+
+            # Print the section only when there are options after filtering
+            if options:
+                print(f"[{section}]")
+                for key, (value, source) in options.items():
+                    if not include_default and source == "default":
+                        continue
+                    print(f"{key} = {value} [{source}]")
+                print()
+        code = buf.getvalue()
+
+        if include_colors:
+            code = pygments.highlight(code=code, formatter=get_terminal_formatter(), lexer=IniLexer())
```

Comment:
one small thing, sorry - i think we should check if we are in terminal or not and only apply color if we are

Reply:
Thanks, makes sense. Used `is_terminal_support_colors` for highlighting.